Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -109,13 +148,53 @@ class Navigator {
     
     var electronicHorizonOptions: ElectronicHorizonOptions? {
         didSet {
-            let nativeOptions: MapboxNavigationNative.ElectronicHorizonOptions?
-            if let electronicHorizonOptions = electronicHorizonOptions {
-                nativeOptions = MapboxNavigationNative.ElectronicHorizonOptions(electronicHorizonOptions)
-            } else {
-                nativeOptions = nil
-            }
-            navigator.setElectronicHorizonOptionsFor(nativeOptions)
+            setupElectronicHorizonOptions()
+        }
+    }
+}
+
+extension Navigator: FallbackVersionsObserver {
+    
+    enum TileVersionState {
+        /// No tiles version switch is required. Navigator has enough tiles for map matching.
+        case nominal
+        /// Navigator does not have tiles on current version for map matching, but TileStore contains regions with required tiles of a different version
+        case shouldFallback([String])
+        /// Navigator is in a fallback mode but newer tiles version were successefully downloaded and ready to use.
+        case shouldReturnToLatest
+    }
+    
+    func onFallbackVersionsFound(forVersions versions: [String]) {
+        switch tileVersionState {
+        case .nominal, .shouldReturnToLatest:
+            tileVersionState = .shouldFallback(versions)
+            
+            guard let fallbackVersion = versions.last else { return }
+            
+            Navigator.shared.restartNavigator(forcing: fallbackVersion)
+            
+            let userInfo: [Navigator.NotificationUserInfoKey: Any] = [
+                .tilesVersionKey: fallbackVersion // internal use only
```

Comment:
It says that is for internal use only, but this key mentioned in public API and the changelog. Didn't follow up all the conversations but this looks suspicious for me.

Reply:
Yeah, the comment might be misleading. This key (and value) used internally for version switching. Even though it is technically accessable for SDK users, we do not encourage them to use or rely on it. Actually, it has next to no practical meaning for them right now, other than curiosity or logging.
In future, we might add more version-related logic for configuring navigation, at which point such data will be handy.

Reply:
Then I think we can hide this information from the public API and changelog. Any changes to this key in the future will be considered as a breaking change, so it would be good to hide the implementation details. The key will still appear in the `userInfo`, but it will be considered a private API because there is no documentation and public keys. We will have more room for future changes related to this functionality. 

@mapbox/navigation-ios What other think about this?

Reply:
Yes, let’s keep it as an undocumented API by removing any mention of it from the changelog and documentation comments on public symbols. (It’s OK to mention it in documentation comments on internal symbols though.)

Reply:
Looks like this is still relevant.